### DISCUSSION TITLE
what is lifetime of commands?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [helix-editor/helix#5784] [Ideas] [answered]
if it exists, then it belongs in QnA
if not, then it belongs to Ideas

i want to know for how long do commands persist? 
like the search command persists even after you move cursor etc, 
i.e. `s` `you` `<ret>` `nn` `jjhjl` `n`
(search for "you", move to next result, move cursor, move to next result)

but there's no indication that the search result is still remembered... 

is it documented somewhere?

currently i can only think that this applies to search command, but maybe there are other commands for which this concept of "memory" or lifetime applies too...

thanks a lot for this awesome editor

**Comment by gabydd** [comment, marked as answer]
Search actually uses a register to store the current search it's the ``/`` (you can see all your registers with ``"``) one so as long as the search register is not written by a new search or yanking some text into it then ``n`` will use what you first searched .

**Reply by goyalyashpal** [reply]
> `n` will use what [`/` register contains]

ahw, that's awesome. wow. thanks a lot for sharing this nugget.

this will fit very well with https://github.com/helix-editor/helix/issues/1596
